cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -487,6 +487,36 @@ private Operation mergeOperationInputOutputs(Operation first, Operation second)
     return result;
   }
 
+  public Set<Operation> getIncomingOperationsForFieldNew(EndPointField destinationField) {
+    if (writeOperations == null) {
+      computeAndValidateFieldLineageInfo(this.operations);
+    }
+
+    Set<String> visitedOperationNames = new HashSet<>();
+    for (WriteOperation write : writeOperations) {
+      if (!write.getDestination().equals(destinationField.getEndPoint())) {
+        continue;
+      }
+
+      List<InputField> inputs = write.getInputs();
+      for (InputField input : inputs) {
+        if (!input.getName().equals(destinationField.getField())) {
+          continue;
+        }
+
+        Map<EndPointField, Set<EndPointField>> result = new HashMap<>();
```

Comment:
is ```result``` still needed? Looks like ```computeIncomingSummaryHelper``` only adds to it, and here it is discarded without reading it.